deepchem/molnet/load_function/uspto_datasets.py

```diff
@@ -4,93 +4,117 @@
-      save_folder = os.path.join(save_folder, img_spec)
-    save_folder = os.path.join(save_folder, str(split))
-
-    loaded, all_dataset, transformers = deepchem.utils.data_utils.load_dataset_from_disk(
-        save_folder)
-    if loaded:
-      return uspto_tasks, all_dataset, transformers
-
-  dataset_file = os.path.join(data_dir,
-                              "2008-2011_USPTO_reactionSmiles_filtered.zip")
-  if not os.path.exists(dataset_file):
-    deepchem.utils.data_utils.download_url(url=USPTO_URL, dest_dir=data_dir)
-
-  # Unzip
-  unzip_dir = os.path.join(data_dir, "2008-2011_USPTO_reactionSmiles_filtered")
-  if not os.path.exists(unzip_dir):
-    deepchem.utils.data_utils.unzip_file(dataset_file, dest_dir=unzip_dir)
-  # Unzipped file is a tap seperated values file (despite the .txt)
-  filename = os.path.join(unzip_dir,
-                          "2008-2011_USPTO_reactionSmiles_filtered.txt")
-  rxns = []
-  from rdkit.Chem import rdChemReactions
-  with open(filename) as tsvfile:
-    reader = csv.reader(tsvfile, delimiter="\t")
-    for ind, row in enumerate(reader):
-      if ind > num_to_load:
-        break
-      if verbose:
-        print("Loading reaction %d" % ind)
-      # The first element in the row is the reaction smarts
-      smarts = row[0]
-      # Sometimes smarts have extraneous information at end of form "
-      # |f:0" that causes parsing to fail. Not sure what this information
-      # is, but just ignoring for now.
-      smarts = smarts.split(" ")[0]
-      rxn = rdChemReactions.ReactionFromSmarts(smarts)
-      rxns.append(rxn)
-  rxn_array = np.array(rxns)
-  # Make up dummy labels since DiskDataset.from_numpy doesn't allow
-  # creation from just features for now.
-  y = np.ones(len(rxn_array))
-  # TODO: This dataset isn't saved to disk so reload doesn't happen.
-  rxn_dataset = DiskDataset.from_numpy(rxn_array, y)
-  transformers = []
-  return uspto_tasks, (rxn_dataset, None, None), transformers
+
+USPTO_MIT_TRAIN = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/USPTO_MIT_train.csv"
+USPTO_MIT_TEST = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/USPTO_MIT_test.csv"
+USPTO_MIT_VALID = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/USPTO_MIT_val.csv"
+
+USPTO_STEREO_TRAIN = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/USPTO_STEREO_train.csv"
+USPTO_STEREO_TEST = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/USPTO_STEREO_test.csv"
+USPTO_STEREO_VALID = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/USPTO_STEREO_val.csv"
+
+
+class _USPTOLoader(_MolnetLoader):
+
+  def __init__(self, *args, subset: str, sep_reagent: bool, **kwargs):
+    super(_USPTOLoader, self).__init__(*args, **kwargs)
+    self.subset = subset
+    self.sep_reagent = sep_reagent
+    self.name = 'USPTO_' + subset
+
+  def create_dataset(self) -> Tuple[Dataset, ...]:
+    #####INCOMPLETE/INCORRECT: I don'd think this is the right way to bypass the splitter!
```

Comment:
Rather than bypass the splitter, you may need to write a new splitter that looks for a `train/val/test` label to do the split on. This could be generically useful for datasets where the sets are already specified.

Reply:
I had thought about this as well, I shall give this a try.